src/pam.rs

```diff
@@ -5,12 +5,14 @@
 
 use pam_types::PamHandle;
 use std::fmt;
+use std::os::raw::c_int;
 
 /// Opaque PAM handle, with additional native methods available via `PamLibExt`.
+#[repr(C)]
```

Comment:
I think this is actually a `rep(transparent)`, and the `PamHandle` type definition may need some adjustment too (a `*const c_void` maybe?).

Reply:
I wasn't aware of `repr(transparent)` it does seem more appropriate.
I changed `PamHandle` to `c_void`. I saw that [the doc](https://doc.rust-lang.org/nomicon/ffi.html#representing-opaque-structs) suggests using an struct containing an empty byte array, but I don't understand how that's any better than `c_void`. And the meaning of `*c_void` is jut more obvious to read.